internal/docstore/mongodocstore/query.go

```diff
@@ -115,3 +137,54 @@ func (it *docIterator) As(i interface{}) bool {
 func (c *collection) QueryPlan(q *driver.Query) (string, error) {
 	return "unknown", nil
 }
+
+// For delete and update queries, limit the number of write actions per RPC, to bound
+// client memory.
+// This is a variable so it can be modified for tests.
+var maxWritesPerRPC = 1000
+
+func (c *collection) RunDeleteQuery(ctx context.Context, q *driver.Query) error {
+	// If there is no limit, use the DeleteMany call.
+	if q.Limit <= 0 {
+		filter, err := c.filtersToBSON(q.Filters)
+		if err != nil {
+			return err
+		}
+		_, err = c.coll.DeleteMany(ctx, filter, nil)
+		return err
+	} else {
+		// If there is a limit, we have to run a query, because DeleteMany doesn't
```

Comment:
If the user provides a limit when they want to "blind" delete on a query, how would they know whether it deletes the documents they wanted? I suggest we might want to not support limit for query-and-writes at the first place.

Reply:
Good point, it does seem weird to do that. I'll make it an error.